common/eth2/src/types.rs

```diff
@@ -1790,12 +1793,12 @@ impl TryFrom<&HeaderMap> for ProduceBlockV3Metadata {
             })?;
         let execution_payload_value =
             parse_required_header(headers, EXECUTION_PAYLOAD_VALUE_HEADER, |s| {
-                Uint256::from_dec_str(s)
+                Uint256::from_str(s)
```

Comment:
To maintain the same behaviour I think we should use `from_str_radix(s, 10)` here. The `from_str` function accepts hex, octal and binary inputs which we should probably reject:

https://docs.rs/ruint/1.12.3/src/ruint/string.rs.html#108-121

Reply:
a7876e7